src/systemtask/SystemTask.cpp

```diff
@@ -342,11 +342,10 @@ void SystemTask::Work() {
           // TODO add intent of fs access icon or something
           break;
         case Messages::OnTouchEvent:
-          if (touchHandler.GetNewTouchInfo()) {
-            touchHandler.UpdateLvglTouchPoint();
+          if (touchHandler.ProcessTouchInfo(touchPanel.GetTouchInfo())) {
+            ReloadIdleTimer();
+            displayApp.PushMessage(Pinetime::Applications::Display::Messages::TouchEvent);
```

Comment:
is the change from "sending all touch events to displayapp including the invalid ones" to "send only valid input events to displayapp" intentional? Please explain why this change is needed as I don't know

Reply:
Previously `UpdateLvglTouchPoints()` was called here when touchinfo is valid. Now DisplayApp updates these points, so in order to update the points using valid data only, the message is only pushed when the data is valid. It could've always been like this, it doesn't make a big difference.